src/modules/wallet/utils.ts

```diff
@@ -1,19 +1,27 @@
-import { Eth } from 'web3x-es/eth'
-import { Address } from 'web3x-es/address'
 import { call, select } from 'redux-saga/effects'
-import { getAddress } from 'decentraland-dapps/dist/modules/wallet/selectors'
-import { createEth } from 'decentraland-dapps/dist/lib/eth'
+import { Eth } from 'web3x-es/eth'
+import { LegacyProviderAdapter } from 'web3x-es/providers'
+import { getData as getBaseWallet } from 'decentraland-dapps/dist/modules/wallet/selectors'
+import { getConnectedProvider, Provider } from 'decentraland-dapps/dist/lib/eth'
+import { Wallet } from 'decentraland-dapps/dist/modules/wallet/types'
+
+export async function getEth(): Promise<Eth> {
+  const provider: Provider | null = await getConnectedProvider()
 
-export function* getCurrentAddress() {
-  const eth: Eth | null = yield call(createEth)
-  if (!eth) {
+  if (!provider) {
     throw new Error('Wallet not found')
   }
 
-  const currentAddress: string = yield select(getAddress)
-  if (!currentAddress) {
-    throw new Error(`Invalid from address: ${currentAddress}`)
+  return new Eth(new LegacyProviderAdapter(provider as any))
+}
+
+export function* getWallet() {
```

Comment:
It seems weird to call it `getWallet` and return `eth` also. Maybe we can rename it to `getEthereum` or something more generic or maybe use two different methods. But it was as it before so I'm not sure if we want to change it.

Reply:
I think it's useful to return both because we're doing 
```
const [wallet, eth]: [Wallet, Eth] = yield getWallet()
```
and using both in a ton of places. It used to be
```
const [address, eth]: [string, Eth] = yield getCurrentAddress()
```

I think splitting it into two is not a bad idea but we'd have to call two methods everywhere

Reply:
ok!